### DISCUSSION TITLE
Inserting a Livewire component into a form makes all the loading buttons spin

### DISCUSSION THREAD

**Discussion by cj-fbr** [filamentphp/filament#10231] [Help] [answered]
Fresh install of...

laravel: 10.35
filament: 3.1.18
livewire: 3.2.6

I've created a LiveWire component to sort a list of items.  It uses an external script to do it's magic.  I want to add this component to a form.  The problem is, when I do this, all the buttons with loading spinners start showing the spinner on page load.  As far as I can tell, it comes from including `@livewireScripts` in the blade file.  The component requires it to work however.  Here is the blade content...

```blade
<div>
    <ul wire:sortable="nonsenseFunction" wire:sortable.options="{ animation: 100 }">
        @foreach (\App\Models\User::all() as $user)
            <li wire:sortable.item="{{ $user->id }}" wire:key="user-{{ $user->id }}">
                <h4>{{ $user->name }} - {{ $user->email }}</h4>
                <button wire:sortable.handle>drag</button>
            </li>
        @endforeach
    </ul>
</div>

@livewireScripts
<script src="https://unpkg.com/@nextapps-be/livewire-sortablejs@0.4.0/dist/livewire-sortable.js"></script>
``` 

Please ignore the utility of this blade.  I just put it together for a quick demo of the problem.

So any thoughts on what is happening here?  What am I doing wrong?

**Comment by ArtMin96** [comment]
Why did you include `@livewireScripts`?

**Reply by cj-fbr** [reply]
Without it, it seems that the external js doesn't actually load.

**Reply by cj-fbr** [reply]
To be clear on this, the code above, as written, works.  That is with the @livewireScripts line in there.  But that causes the loading spinners to be always on.  If I remove the @livewireScripts line, the spinner behave, but the code doesn't work (no drag & drop sorting functionality).

**Reply by leandrocfe** [reply]
try to include assets using this
https://filamentphp.com/docs/3.x/support/assets

**Reply by cj-fbr** [reply]
No luck.  Same effect.  I need to have the @livewireScripts line there in order for it to run properly.  

Perhaps related: In the filament docs about upgrading there is this note...
`The @filamentScripts and @filamentStyles Blade directives must be added to your Blade layout file/s. Since Livewire v3 no longer uses similar directives, you can replace @livewireScripts with @filamentScripts and @livewireStyles with @filamentStyles.`

https://filamentphp.com/docs/3.x/notifications/upgrade-guide#new-filamentscripts-and-filamentstyles-blade-directives

But it appears livewire still uses @livewireScripts (and @livewireStyles).
https://livewire.laravel.com/docs/installation#manually-including-livewires-frontend-assets

**Reply by leandrocfe** [reply]
I didn't understand why you need to include it. Anyway, you could try to add it via hooks..?
https://filamentphp.com/docs/3.x/support/render-hooks#registering-render-hooks

**Reply by cj-fbr** [reply, marked as answer]
I don't understand why either, but I have found a solution:  @assets tag(s)

I'm not sure how I ended up on the @livewireScripts route, but after going through the docs a bit more, this is the obvious use case.
https://livewire.laravel.com/docs/javascript#loading-assets

So the new code snippet that works is simply...

```blade
@assets
<script src="https://unpkg.com/@nextapps-be/livewire-sortablejs@0.4.0/dist/livewire-sortable.js" defer></script>
@endassets

<div>
    <ul wire:sortable="nonsenseFunction" wire:sortable.options="{ animation: 100 }">
        @foreach (\App\Models\User::all() as $user)
            <li wire:sortable.item="{{ $user->id }}" wire:key="user-{{ $user->id }}">
                <h4>{{ $user->name }} - {{ $user->email }}</h4>
                <button wire:sortable.handle>drag</button>
            </li>
        @endforeach
    </ul>
</div>

```

**Reply by Khant-Nyar** [reply]
Didi I need to asset the alpine?

**Reply by krv** [reply]
Thank you that fixed it for me too, also with the livewire-sortable plugin
